### ISSUE TITLE
Sync namespaces dont work

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#70123]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Sync-namespaces-dont-work/10473603)._

---
In the latest versions of VS 2022 **"Sync Namespaces"** dont work. If i have a file with a namespace that dont match the folder structure, I expect **"Sync Namespaces"** to rename my namespace. But it always report "No namespaces needed updating."

---
#### Alison Jin [MSFT] on 9/24/2023, 07:52 PM: 

(private comment, text removed)
#### Björn Holmgren on 9/25/2023, 01:23 AM: 

(private comment, text removed)
#### Feedback Bot on 9/25/2023, 01:34 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by TobiasKoller** [comment]
I've the same problem. Copied a folder with some cs-files to a new project, run Sync namespaces and it tells me that no namespace has to be fixed.

**Comment by BillyMartin1964** [comment]
Yes. I have this issue with one whole Maui Project, while other projects in the same solution work fine.

**Comment by kumarchhajed** [comment]
Doesn't work for Class Library Project as well. VS 2022 17.9.3

**Comment by XtremeOwnageDotCom** [comment]
Also- to add to this- if you just remove the namespace from a file- it doesn't fix that either.


I had a smart idea of replace ^namespace .* $ with "", and then let the namespace fixer fix it. Nope. Doesn't handle that either.

**Comment by sjain882** [comment]
![Image](https://github.com/user-attachments/assets/d1d738d4-2d36-4eb6-87b4-f7820a6cbb4c)
Seeing this on 17.12.3

**Comment by CyrusNajmabadi** [comment]
@sjain882 That is unrelated to this issue.

**Comment by OFark** [comment]
This is still happening in Visual Studio 2026, one project (Class library) out of ten won't fix namespaces, there's no .editorconfig in that folder root, the only thing in the project file is the package references. The project is SDK style. It still happens if I comment out all of the packages.
If I add a .editorconfig with the project and set `dotnet_diagnostic.IDE0130.severity = warning` then for the files I have open Sync Namespace will work, and I do get the warnings in the Error List window.
In Tools -> Options -> Languages -> C# -> More Settings -> Advanced -> `Run background code analysis for:` "Entire solution" and the same for `Show compiler errors and warnings for:` allows the Sync Namespaces to work, but I don't understand why it would need this setting for just this project?
